### ISSUE TITLE
Bad commits everywhere

### ISSUE THREAD

**Issue by klauer** [pcdshub/push-to-epics-slac-tools#1]
## Background

There are a large number of commits that are missing authors or e-mail addresses due to poor practices during the svn-git migration of modules.

GitHub will reject the entire `push --mirror` (or `push --tags --all`) in the event that such commits exist:

```bash
$ git push github-slac --tags
Enumerating objects: 297, done.
Counting objects: 100% (297/297), done.
Delta compression using up to 16 threads
Compressing objects: 100% (149/149), done.
Writing objects: 100% (284/284), 35.71 KiB | 8.93 MiB/s, done.
Total 284 (delta 181), reused 214 (delta 127), pack-reused 0
remote: error: object f0c2ef8e88f59a8fc332bda9a5200193e3bee990: badDate: invalid author/committer line - bad date
remote: fatal: fsck error in packed object
error: remote unpack failed: index-pack abnormal exit
To github.com:slac-epics/genSub
 ! [remote rejected] End-of-run-August-2012 -> End-of-run-August-2012 (failed)
 ! [remote rejected] GENSUB_1_6 -> GENSUB_1_6 (failed)
 ! [remote rejected] R1.6.1.2-0.0.0 -> R1.6.1.2-0.0.0 (failed)
...
```

A workaround, taking `genSub`, for example:

```bash
$ git fsck --no-progress
error in commit 0b587008b4fcbfc95c4dccb359182e377869d17c: badDate: invalid author/committer line - bad date
error in commit f0c2ef8e88f59a8fc332bda9a5200193e3bee990: badDate: invalid author/committer line - bad date

# Push tags without those commits
$ git push github-slac  $(git tag --no-contains f0c2ef8e88f59a8fc332bda9a5200193e3bee990 --no-contains 0b587008b4fcbfc95c4dccb359182e377869d17c)

# Push branches without those commits
$(git branch --no-contains f0c2ef8e88f59a8fc332bda9a5200193e3bee990 --no-contains 0b587008b4fcbfc95c4dccb359182e377869d17c --no-merged)
```

I have a git alias that does this, but it's not set up for multiple bad commits:

```gitconfig
[alias]
   push-to-gh-slac-without-commit="!func() {  git push github-slac $(git tag --no-contains $1); git push github-slac $(git branch --no-contains $1 --no-merged); }; func"
```

## In short

* Automate the `fsck` and the push to exclude bad histories by default
* (Perhaps rewrite history at some point, if TID allows for it...)
